src/contexts/authProvider.tsx

```diff
@@ -31,43 +27,30 @@ const AuthenticateProvider: React.FC = ({ children }) => {
   const [role, setRole] = useState<RoleOptions>('unconfirmed');
 
   const updateCredentials = () => {
-    verifyAuthentication().then((res) => {
-      if (res) {
-        const userRole = getRole() as RoleOptions;
+    getTokenInfo()
+      .then((res) => {
+        const userRole = res.role as RoleOptions;
+        monitorRefreshToken(res.exp);
         setRole(userRole);
-        setAuthenticated(res);
+        setAuthenticated(true);
         setValidating(false);
-      } else {
-        setValidating(true);
+      })
+      .catch(() => {
         setRole('unconfirmed');
         setAuthenticated(false);
         setValidating(false);
-      }
-    });
+      });
   };
 
   // function for checking the refreshToken after it expired to set the user to not authenticated when inactive
-  const monitorRefreshToken = () => {
-    // small offset to handle if renewal was valid but it took time for the tokens to be retrieved ans sat
-    const offset = 10 * 1000;
-    const exp = findRefreshTokenExpInMs();
-    if (exp === undefined) {
-      // refreshToken is expired
-      setAuthenticated(false);
-      return;
-    }
-    setTimeout(monitorRefreshToken, exp + offset);
+  const monitorRefreshToken = async (exp: number) => {
```

Comment:
This should work pretty decent. I am however not 100% sure what would happen if it starts multiple `setTimeout`'s. This could potentially happen since `updateCredentials` is exposed and called from a couple of other places. I know `setTimeout` can be cancelled with `clearTimeout` by the return value. We should make sure that this won't suddenly start spamming requests to try to renew the token